mt2.tex

```diff
@@ -18,4 +22,373 @@
+	\begin{frame}{Practice Problem Solution}
+	\begin{enumerate}[(i)]
+	    \item We were given the DC input, $u^* = 0$ \pause \\
+	    \item $cos(x^*)$ = 0, which means that $x^* = k\frac{\pi}{2}$ for $k \in \{...-2,-1,1,2,...\}$. We will choose $x^* = \frac{\pi}{2}$ \pause \\
+	    \item Let $x_l(t) = x(t) - \frac{\pi}{2}$ and $u_l(t) = u(t) - 0$.
+	    By plugging in we get: $\frac{dx_l(t)}{dt} = cos(x_l(t) + \frac{\pi}{2}) + u_l(t)$ \pause \\
+	    \item We assume that $u_l(t)$ is small.\pause \\
+	    \item Linearize the ODE: $cos(x_l(t) + \frac{\pi}{2}) \approx cos(\frac{\pi}{2}) -sin(\frac{\pi}{2})x_l(t)$. \pause
+	    \item Plug (v) back into ODE: 
+	    $\frac{dx_l(t)}{dt} = -x_l(t) + u_l(t)$ 
+	\end{enumerate}
+	 \pause 
+	 We see that our assumption that  $x_l(t)$ is small is indeed satisfied as we will have a $e^{-t}$ term in the solution which means that $x_l(t)$ will decay.\\ \pause
+	What if we had chosen a different DC Operating point, say $-\frac{\pi}{2}$? When we linearize the system we see that the solution will "explode" around that particular DC operating point.
+	\end{frame}
+
+	\begin{frame}{Linearization of Vector Functions}
+	What if we had $\frac{d\vec{x}}{dt} = \vec{f}(\vec{x}, \vec{u})$ ? We will see that the process is very similar to the scalar case we just analyzed!\\ \pause
+	First, let's see how to linearize $\vec{f}(\vec{x})$ around a DC operating point $\vec{x}^*$. Where $\vec{f} \in \R^{nx1}$ is a vector of scalar functions. \\\pause
+	The idea is to linearize individually each one of the $f_i$ around the DC operating point. \\\pause
+	For example: $f_1 \approx f_1(\vec{x}^*) + \frac{\partial f_1}{\partial x_1}(\vec{x}^*)(x_1 - x_1^*) + ... +  \frac{\partial f_n}{\partial x_1}(\vec{x}^*)(x_n - x_n^*)$\\\pause
+	Repeating this for all $n$ functions in $\vec{f}$ we see we get a system of scalar, linearized, multivariate functions which makes you think, wouldn't it be nice to express it in a shorthand matrix notation?
+	\end{frame}
+
+	\begin{frame}{Jacobian Matrix}
+	We can use the Jacobian to express everything nicely and neatly. \\
+	The Jacobian is the name given to the matrix of partial derivatives of $\vec{f}$, and it is denoted by $J_{\vec{x}}$ or $\nabla_{\vec{x}}\vec{f}$. \\ \pause
+	Continuing from the previous slide we have: \\
+	\[
+	\begin{bmatrix} f_1(\vec{x}) \\ \vdots \\f_n(\vec{x}) \end{bmatrix} \approx 
+	\begin{bmatrix} f_1(\vec{x}^*) \\ \vdots \\f_n(\vec{x}^*) \end{bmatrix} + 
+	\begin{bmatrix} \frac{\partial f_1}{\partial x_1}(\vec{x}^*)& \hdots& \frac{\partial f_1}{\partial x_n}(\vec{x}^*) \\ \vdots & \ddots &\vdots \\ \frac{\partial f_n}{\partial x_1}(\vec{x}^*)& \hdots &  \frac{\partial f_n}{\partial x_n}(\vec{x}^*) \end{bmatrix}
+	(\vec{x} - \vec{x}^*)
+	\]
+	\\
+
+	\end{frame}
+
+	\begin{frame}{Linearization with Jacobians Example}
+	    \begin{align*}
+	    \text{Linearize}  \hspace{0.2 cm} 
+	     \vec{f}(\vec{x}(t))  = 
+	    \begin{bmatrix}
+	    sin(x_1(t)*x_2(t)) + 2x_1(t)x_3^2(t)\\
```

Comment:
\sin and \cos instead of sin and cos in this line and the one below

Reply:
Also, this could just be me but it feels like a lot more algebra then the class normally requires.